@app/server/src/plugins/CreateUploadUrlPlugin.ts

```diff
@@ -0,0 +1,37 @@
+import { makeExtendSchemaPlugin, gql } from "graphile-utils";
+import uuidv4 from "uuid/v4";
+import * as aws from "aws-sdk";
+import { awsRegion, uploadBucket } from "@app/config";
+
+const CreateUploadUrlPlugin = makeExtendSchemaPlugin(() => ({
+  typeDefs: gql`
+    extend type Mutation {
+      """
+      Get a signed URL for uploading files. It will expire in 60 seconds.
+      """
+      createUploadUrl(contentType: String): String!
+    }
+  `,
+  resolvers: {
+    Mutation: {
+      async createUploadUrl(_query, args, _context, _resolveInfo) {
+        const { contentType } = args;
+        const s3 = new aws.S3({
+          region: awsRegion,
+          signatureVersion: "v4",
+        });
+        const params = {
+          Bucket: uploadBucket,
+          ContentType: contentType,
+          Key: uuidv4(), // randomly generated file name
+          Expires: 60, // signed URL will expire in 60 seconds
+          ACL: "public-read", // uploaded file will be publicly readable
+        };
+        const signedUrl = await s3.getSignedUrlPromise("putObject", params);
```

Comment:
There's no auth check, so anyone could upload any file and have it publicly accessible, and we'd be left footing the bill. Let's at least require that they're logged in. I wonder if we should just store to `/avatars/${user_id}` that way there's at most one avatar per user. We wouldn't necessarily know if it changed though...